ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/StoreBackedCommandExecutor.scala

```diff
@@ -56,7 +57,13 @@ private[apiserver] final class StoreBackedCommandExecutor(
   ): Future[Either[ErrorCause, CommandExecutionResult]] = {
     val interpretationTimeNanos = new AtomicLong(0L)
     val start = System.nanoTime()
+    val coids = commands.commands.commands.toSeq.foldLeft(Set.empty[Value.ContractId]) {
```

Comment:
Yes, this only applies to the top level exercises and the contract arguments. This optimises for the standard case where command batching is used. All performance related tests heavily batch top level exercises together, so this change does optimise for such scenarios.

Reply:
My point was that even if there is one contract lookup in those exercises, then we are back to the original issue: sequential singular lookups.
Interesting you say performance tests. Do we have command batching in real life scenarios too? I am curious since we have not much optimizations for boosting batched commands. Is command batching something we recommend our clients to do?